### PR TITLE
Add Subscription Service API

### LINKED ISSUE #20: Feature: Add Subscription Service API
- Read All
- Read One
- Update One
- Delete One
- Create One
    - Before passing the `expiry` parameter to the backend, convert the `expiry` datetime from the user's system timezone to UTC using [moment.js](https://momentjs.com/docs/). Likewise for updating subscription.

### PR DESCRIPTION
**Pull request by elizabethhky**
## Changes
- Add Modules
  - subscription
- Add SubscriptionService API
 
## Testing
- Click on buttons 'Create Subscription', 'Get All Subscriptions' etc. underneath Subscription section on the tests page
- Created dummy data in Test page

## Issues
- Resolves #20

### CODE REVIEW

File: src/modules/subscription/types.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+export interface SubscriptionData {
+  id: number;
+  name: string;
+  description: string;
+  accessRights: string[];
```
Comment (ShenyiCui) [inline src/modules/subscription/types.ts:5]: same as the comment mentioned in #21

File: src/pages/Test/Test.tsx [hunk 2]
```diff
@@ -12,6 +14,16 @@ const Input = styled('input')({
   display: 'none',
 });
 
+const createSubscriptionData: CreateSubscriptionData = {
+  name: 'Adobe Photoshop',
+  description: 'For photo editing',
+  accessRights: ['Admin', 'Teacher'],
```
Comment (ShenyiCui) [inline src/pages/Test/Test.tsx:20]: Same as #21 comment regarding creating a constant for Roles

File: src/pages/Test/Test.tsx [hunk 3]
```diff
@@ -12,6 +14,16 @@ const Input = styled('input')({
...
+  credentials: 'test123',
+  expiry: '2023-06-01',
```
Comment (ShenyiCui) [inline src/pages/Test/Test.tsx:22]: @jasonqiu212  will push a fix later for the expiry date. The date it accepts now will be slightly different. Ask @jasonqiu212 for the new date requirements in the backend and make the necessary changes

**Review by ShenyiCui — CHANGES_REQUESTED** [review]
sorry, i accidentally approved it, please make the required changes first, thanks!

**Review by jasonqiu212 — CHANGES_REQUESTED** [review]
some minor comments! 

as for the type of the datetime, i'm not too sure if using type `Moment` is the way to go. Since ultimately, when you pass the date to the backend, it will converted into a string. I was thinking of creating a custom `dateString` type to handle this. lemme consult with Shenyi

**Pull request merged**